### ISSUE TITLE
Allow non opt-in user to use rewrite in-place AI tools 

### ISSUE THREAD

**Issue by yrliou** [brave/brave-browser#37260]
https://github.com/brave/brave-browser/issues/36865 only provides rewrite in-place for users who already opted in, otherwise it would go through the sidebar flow first (which would show the opt-in message).
@bbondy has brought up that we want to allow them to do this by triggering some UI during the PR review, https://github.com/brave/brave-core/pull/22639#discussion_r1536135480.
This would need some designs for this UI, cc @aguscruiz

**Comment by aguscruiz** [comment]
@yrliou @bbondy just checking to see if I understood, what we need here is just a way for a new user to accept the agreement if it's a first use? Without needing to open the panel, right?

**Comment by bbondy** [comment]
yep, or if you think it should just open the panel, then let us know. But is it a bit weird if it opens the panel since the feature doesn't route to the panel?

**Comment by aguscruiz** [comment]
Gotcha. Ok yeah I think we need a browser level message instead. I suggest we treat it like one of those browser native notifications. What do you think?

I added the design here:
https://www.figma.com/file/m0Gdbf0wtqyfEFGm32VLLc/Leo?type=design&node-id=2887%3A64446&mode=design&t=4AkIWMMdo6FcVgZj-1

**Note:** I don't know if the copy I have is the latest, but don't take it as the final version just in case. We should use the same one that's in produciton.

**Comment by aguscruiz** [comment]
But now that I think about this, we need to create some sort of UI for all the error messages and limitations the user may bump into when working with this. 

@bbondy do you think opening the panel then would be good? Otherwise we may need to invest a bit of time to develop something like the floating Leo idea that can be open and closed right there.

What do you think?

![image](https://github.com/brave/brave-browser/assets/15734630/7266f7d4-0c71-4d18-a076-b3bb5a7739b6)

**Comment by aguscruiz** [comment]
Now that I think about this, we need to create some sort of UI for all the error messages and limitations the user may bump into when working with this.

@bbondy @yrliou do you think opening the panel then would be good in that case? Otherwise we may need to invest a bit of time to develop one of the two other versions of Leo I have on the designs

https://www.figma.com/file/m0Gdbf0wtqyfEFGm32VLLc/Leo?type=design&node-id=2897%3A67137&mode=design&t=YlzjCSviObwEo0Wj-1


### Floating Leo alternative
![image](https://github.com/brave/brave-browser/assets/15734630/22ae9e14-e3fc-4c51-888d-f0e86ee0386b)

### Leo writer alternative
![image](https://github.com/brave/brave-browser/assets/15734630/a9090bf9-24df-43a7-9e95-df4d9c14f02a)

**Comment by yrliou** [comment]
Just want to add that currently both errors and for users who haven't opted in will use the side panel (which will use in-place next time). So it mainly depends on whether we would like to have another set of UI developed outside of side panel for these, which I don't have a strong opinion on either choice.

**Comment by mattmcalister** [comment]
Closing as stale.
